docs/nextflow_schema/schema_specification.md

```diff
@@ -310,17 +311,16 @@ Example usage is as follows:
 The available `format` types are below:
 
 `file-path`
-: States if the provided value is a file. Does not check its existence.
+: States and checks if the provided value is a file. Does not check its existence.
```

Comment:
@nvnieuwk  How does it check if it's a file, if it's not looking to see if it exists? Surely any string would validate?

Reply:
Good point it mainly checks if the path isn't a directory instead of checking if it's a file. You can see the code here: https://github.com/nextflow-io/nf-validation/blob/master/plugins/nf-validation/src/main/nextflow/validation/FormatValidators/FilePathValidator.groovy, I'll adjust the docs to better describe what it does here :)

Reply:
Thanks! I'd like to dig into this a bit more, curious what `isDirectory()` does if it's not checking if it exists or not (how can it check the path properties if it doesn't exist). I'm not sure how you can tell if `/foo/bar` is a file or a directory based on just the string?

@bentsherman any insight into how [Nextflow file `isDirectory()`](https://github.com/nextflow-io/nf-validation/blob/1de02dd655335c4eec34243f303b30d4e3e3e862/plugins/nf-validation/src/main/nextflow/validation/FormatValidators/FilePathValidator.groovy#L12-L13C18) is working when not set to check if the file exists?

Reply:
It will just return `false` if the directory does not exist.

Reply:
Nicolas is right. If you're ever curious you can just check the Java [Path docs](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html).

Reply:
Well, in this case it's defined under [Files](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html), but those two classes cover just about everything

Reply:
Ok, then this is maybe a problem. As it means that the `exists` flag is redundant for `"format": "directory"` - it's impossible to specify a non-existent directory.

This is a very common scenario, eg. `--outdir` which will create a directory if it doesn't already exist. But we want to specify in the schema that it's for a directory still.

See where I'm going here?